Lib/ufo2ft/outlineCompiler.py

```diff
@@ -410,6 +410,134 @@ def setupTable_cmap(self):
             # update tables registry
             cmap.tables = [cmap4_0_3, cmap4_3_1, cmap12_0_4, cmap12_3_10]
 
+    def _calcCodepageRanges(self):
+        codepageRanges = set()
+
+        unicodes = [i for i in self.unicodeToGlyphNameMapping.keys() if i is not None]
+
+        hasAscii = len([u for u in unicodes if 0x20 <= u <= 0x7E]) == 95
+        hasLineart = False
+        hasRadical = False
+        hasSummation = False
+        hasCP708 = False
+        hasCP737 = False
+        hasCP775 = False
+        hasCP852 = False
+        hasCP857 = False
+        hasCP860 = False
+        hasCP861 = False
+        hasCP862 = False
+        hasCP863 = False
+        hasCP864 = False
+        hasCP865 = False
+        hasMac = False
+        hasCP855 = 0
+        hasCP866 = 0
+        hasCP869 = 0
+        for code in unicodes:
+            if code == 0x00DE and hasAscii:
```

Comment:
all this magic constant hex digits, would be nice to have comments showing what characters they refer to.
maybe you could use `byteord('Þ')` (defined in `fontTools.misc.py23`) so it's clear what these characters are.